test/spec.git-parser.js

```diff
@@ -9,42 +8,383 @@ describe('git-parser stash show', () => {
+    gitLog += 'Date:   Wed Jun 4 19:49:17 2014 -0700\n'
+    gitLog += 'signed commit'
 
-      if(refs.length != 0) {
-        throw new Error('Failed to parse git log without branches.');
-      }
+    expect(gitParser.parseGitLog(gitLog)[0]).to.eql({
+      authorEmail: "test@example.com",
+      authorName: "Test Ungit",
+      fileLineDiffs: [],
+      isHead: true,
+      message: "",
+      parents: [
+        "d58c8e117fc257520d90b099fd2c6acd7c1e8861"
+      ],
+      refs: [
+        "HEAD",
+        "refs/heads/git-parser-specs"
+      ],
+      sha1: "37d1154434b70854ed243967e0d7e37aa3564551",
+      signatureDate: "Wed Jun  4 19:49:17 2014 PDT using RSA key ID 0AAAAAAA",
+      signatureMade: '"Test ungit (Git signing key) <test@example.com>"'
     });
   });
+  it('parses the git log', () => {
+    var gitLog = 'commit 37d1154434b70854ed243967e0d7e37aa3564551 d58c8e117fc257520d90b099fd2c6acd7c1e8861 (HEAD -> refs/heads/git-parser-specs)\n'
+    gitLog += 'Author:     Test ungit <test@example.com>\n'
+    gitLog += 'AuthorDate: Fri Jan 4 14:03:56 2019 +0100\n'
+    gitLog += 'Commit:     Test ungit <test@example.com>\n'
+    gitLog += 'CommitDate: Fri Jan 4 14:03:56 2019 +0100\n'
+    gitLog += '\n'
+    gitLog += '    submodules parser\n'
+    gitLog += '\n'
+    gitLog += '32\t0\ttest/spec.git-parser.js\n'
+
+    expect(gitParser.parseGitLog(gitLog)[0]).to.eql(
+      { 
+        refs: [ 'HEAD', 'refs/heads/git-parser-specs' ],
+        fileLineDiffs: [
+          [32, 0, "Total"],
+          [32, 0, "test/spec.git-parser.js", "text"]
+        ],
+        sha1: '37d1154434b70854ed243967e0d7e37aa3564551',
+        parents: [ 'd58c8e117fc257520d90b099fd2c6acd7c1e8861' ],
+        isHead: true,
+        authorName: 'Test ungit',
+        authorEmail: 'test@example.com',
+        authorDate: 'Fri Jan 4 14:03:56 2019 +0100',
+        committerName: 'Test ungit',
+        committerEmail: 'test@example.com',
+        commitDate: 'Fri Jan 4 14:03:56 2019 +0100',
+        message: 'submodules parser',
+      }
+    );
+  });
 });
 
 describe('git-parser submodule', () => {
   it('should work with empty string', () => {
     const gitmodules = "";
     const submodules = gitParser.parseGitSubmodule(gitmodules);
-    expect(submodules).to.be.an('object').and.to.be.empty();
+    expect(submodules).to.eql({})
   });
   it('should work with name, path and url', () => {
     const gitmodules = '[submodule "test1"]\npath = /path/to/sub1\nurl = http://example1.com';
     const submodules = gitParser.parseGitSubmodule(gitmodules);
     expect(submodules.length).to.be(1);
-    expect(submodules[0].name).to.be('test1');
-    expect(submodules[0].path).to.be(path.join(path.sep, 'path', 'to', 'sub1'));
-    expect(submodules[0].url).to.be('http://example1.com');
+    expect(submodules[0]).to.eql({
+      name: "test1",
+      path: "/path/to/sub1",
+      rawUrl: "http://example1.com",
+      url: "http://example1.com"
+    });
```

Comment:
👍 I think this is a cleaner comparison!! I didn't know you could compare objects like this. I wonder if order matters or fully recursive...